snapcraft/tests/test_sources.py

```diff
@@ -68,6 +71,89 @@ def test_pull_tarball_must_download_to_sourcedir(self, mock_prov):
         with open(os.path.join(dest_dir, tar_file_name), 'r') as tar_file:
             self.assertEqual('Test fake compressed file', tar_file.read())
 
+    def test_checksum(self):
```

Comment:
maybe I am confused by `check_checksum_determine_format` and a better method name is needed here.